src/components/button/Button.base.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+import Grid, { Spacing, TSpacingTokensSymmetric } from '../../foundations/layout';
+import Shape from '../../foundations/shape';
+import Icon, { TIconSize } from '../icon';
+import Text, { TTextSizeToken } from '../text';
+
+import { PButton } from './Button.props';
+
+const ButtonBase: React.FC<PButton> = ({
+    label,
+    leadingIcon,
+    trailingIcon,
+    size,
+    width,
+    ...rest
+}: PButton) => {
+    let labelSize: TTextSizeToken = 100;
+    let iconSize: TIconSize = 18;
+    let height = 40;
+
+    const spacing: TSpacingTokensSymmetric = {
+        vertical: 0,
+        horizontal: 125,
+    };
+
+    switch (size) {
+        case 'large':
+            labelSize = 200;
+            iconSize = 24;
+            height = 48;
+            spacing.horizontal = 150;
+            break;
+        case 'small':
+            labelSize = 75;
+            iconSize = 14;
+            height = 32;
+            // line-height on text is 16, so there is no need to adjust paddings
+            spacing.horizontal = 100;
+            break;
+        case 'medium':
+        default:
+    }
+
+    return (
+        <Shape
+            component={'button'}
+            borderRadius={4}
+            width={width === 'full' ? '100%' : width}
+            height={height}
+            {...rest}
+        >
+            <Grid
+                row
+                alignment={'center'}
+                justify={'center'}
+                padding={Spacing.symmetric(spacing)}
+                flex={1}
+            >
```

Comment:
Grid defaults to a `div`, but div's can't exist inside `buttons`, we should set this to a `span` here.

I'm still not entirely happy with the extra container element needed to accomplish this. I would prefer to not need extra containers on base components like buttons.